pkg/service/service.go

```diff
@@ -119,6 +122,17 @@ func (s *deviceService) Run() error {
 		},
 	})
 
+	// When the number of runtime.GOMAXPROCS is less than 20, the pool concurrency is configured to 20 to increase performance
+	poolSize := runtime.GOMAXPROCS(0)
```

Comment:
you can use the configurable `AsyncBufferSize` as the pool size
https://github.com/edgexfoundry/device-sdk-go/blob/97b36af7baff2da76e29dc2359062fa76ab64ed3/internal/config/types.go#L51

the default value is 16
https://github.com/edgexfoundry/edgex-go/blob/17930b14352cbfb7930b1ac1271c629a4bf190e4/cmd/core-common-config-bootstrapper/res/configuration.yaml#L135

Reply:
Although taking the maximum of GOMAXPROCS and AsyncBufferSize could get the best performance, it would make users lost the configurable mechanism.
Sometimes, they don't want to give a single service too much resource.